spec.md

```diff
@@ -1495,6 +1638,108 @@ For success responses, the following fields are defined:
 }
 ```
 
+## Adapting Service Binding Credentials
+
+If the endpoints of a Service Instance are from an Application point of view
+different to the endpoints provided in the binding credentials, then the
+binding credentials MUST be adapted before the Platform provides them to
+the Application.
+
+The Platform requests the corrected credentials from the Service Broker by
+providing the original credentials and a table that maps the original
+endpoints to the new endpoints.
+
+### Request
+
+##### Route
+
+`POST /v2/service_instances/:instance_id/service_bindings/:binding_id/adapt_credentials`
+
+`:instance_id` MUST be the ID of a previously provisioned Service Instance.
+
+`:binding_id` MUST be the ID of a previously provisioned Service Binding for that
+instance.
+
+#### Body
+| Request Field | Type | Description |
+| --- | --- | --- |
+| credentials* | object | The free-form hash of credentials that the Platform has received from the Service Broker. |
+| endpoint_mappings* | array of [EndpointMapping](#endpoint-mapping-object) objects | A list of endpoint mapping. |
+
+\* Headers with an asterisk are REQUIRED.
+
+##### Endpoint Mapping Object
+
+| Request Field | Type | Description |
+| --- | --- | --- |
+| source* | [Endpoint](#endpoint-object) object | The endpoint on the service side. |
+| target* | [Endpoint](#endpoint-object) object | The mapped endpoint on the application side. |
+
+The `ports` arrays in both endpoints MUST be of the same size and the order of
```

Comment:
Do you prefer to have this same index for host and port or another object, like: `[{ "host": "appnethost", "port": 9876}]`?